lib/core/proxy.c

```diff
@@ -700,6 +700,7 @@ static h2o_httpclient_head_cb on_connect(h2o_httpclient_t *client, const char *e
         }
     }
     self->client->informational_cb = on_1xx;
+    self->src_req->proxy_ssl = client->get_socket(client)->ssl;  // used in logging upstream ssl info
```

Comment:
I'm not sure if this is going to work. The problem is that the socket connecting to upstream might get discarded before the logger tries to read the value through `h2o_req_t::proxy_ssl`.

I think we'd have to collect the values and record them, like we do with `h2o_httpclient_t::timings`.

Reply:
Got it and pushed follow-up.
Have a nice day!